node/src/rpc.rs

```diff
@@ -5,60 +5,101 @@
-	C: Send + Sync + 'static,
-	C::Api: substrate_frame_rpc_system::AccountNonceApi<B, AccountId, AccountIndex>,
-	C::Api: pallet_transaction_payment_rpc::TransactionPaymentRuntimeApi<B, Balance>,
-	C::Api: assets_runtime_api::AssetsRuntimeApi<B, CurrencyId, AccountId, Balance>,
-	C::Api: crowdloan_rewards_runtime_api::CrowdloanRewardsRuntimeApi<B, AccountId, Balance>,
-	C::Api: pablo_runtime_api::PabloRuntimeApi<B, PoolId, CurrencyId, Balance>,
-	C::Api: BlockBuilder<B>,
-	P: TransactionPool + 'static,
+	RuntimeApi:
+		ConstructRuntimeApi<OpaqueBlock, FullClient<RuntimeApi, Executor>> + Send + Sync + 'static,
+	RuntimeApi::RuntimeApi:
+		BaseHostRuntimeApis<StateBackend = StateBackendFor<FullBackend, OpaqueBlock>>,
+	StateBackendFor<FullBackend, OpaqueBlock>: StateBackend<BlakeTwo256>,
+	Executor: NativeExecutionDispatch + 'static,
+	FullClient<RuntimeApi, Executor>: ProvideRuntimeApi<OpaqueBlock>
+		+ HeaderBackend<OpaqueBlock>
+		+ HeaderMetadata<OpaqueBlock, Error = sp_blockchain::Error>
+		+ 'static
+		+ Send
+		+ Sync
+		+ Sized,
+	<FullClient<RuntimeApi, Executor> as ProvideRuntimeApi<OpaqueBlock>>::Api:
+		BaseHostRuntimeApis<StateBackend = StateBackendFor<FullBackend, OpaqueBlock>>
+			+ ExtendWithAssetsApi<RuntimeApi, Executor>
+			+ ExtendWithCrowdloanRewardsApi<RuntimeApi, Executor>
+			+ ExtendWithPabloApi<RuntimeApi, Executor>
+			+ ExtendWithLendingApi<RuntimeApi, Executor>,
 {
 	use pallet_transaction_payment_rpc::{TransactionPayment, TransactionPaymentApi};
 	use substrate_frame_rpc_system::{FullSystem, SystemApi};
 
 	let mut io = jsonrpc_core::MetaIoHandler::default();
-	let FullDeps { client, pool, deny_unsafe } = deps;
 
-	io.extend_with(SystemApi::to_delegate(FullSystem::new(client.clone(), pool, deny_unsafe)));
+	io.extend_with(SystemApi::to_delegate(FullSystem::new(
+		deps.client.clone(),
+		deps.pool.clone(),
+		deps.deny_unsafe.clone(),
+	)));
+
+	io.extend_with(TransactionPaymentApi::to_delegate(TransactionPayment::new(
+		deps.client.clone(),
+	)));
 
-	io.extend_with(TransactionPaymentApi::to_delegate(TransactionPayment::new(client.clone())));
+	<FullClient<RuntimeApi, Executor> as ProvideRuntimeApi<OpaqueBlock>>::Api::extend_with_assets_api(
+		&mut io,
+		deps.clone(),
+	);
 
-	io.extend_with(AssetsApi::to_delegate(Assets::new(client.clone())));
+	<FullClient<RuntimeApi, Executor> as ProvideRuntimeApi<OpaqueBlock>>::Api::extend_with_crowdloan_rewards_api(
+		&mut io,
+		deps.clone(),
+	);
 
-	io.extend_with(CrowdloanRewardsApi::to_delegate(CrowdloanRewards::new(client.clone())));
+	<FullClient<RuntimeApi, Executor> as ProvideRuntimeApi<OpaqueBlock>>::Api::extend_with_pablo_api(
+		&mut io,
+		deps.clone(),
+	);
 
-	io.extend_with(PabloApi::to_delegate(Pablo::new(client)));
+	<FullClient<RuntimeApi, Executor> as ProvideRuntimeApi<OpaqueBlock>>::Api::extend_with_lending_api(
+		&mut io,
+		deps.clone(),
+	);
+
+	// io.extend_with(PabloApi::to_delegate(Pablo::new(client)));
```

Comment:
pablo will be releases soon, earlier than lending, so may be return it back?

Reply:
I think keeping all the RPC apis implemented this way is ideal - keeps it consistent and allows us to remove/add rpcs as we need to

Reply:
see latest commits for boilerplate impl